test/tutorial/test_rate_constants_user_defined_by_hand.cpp

```diff
@@ -0,0 +1,211 @@
+                   .rate_constant(TunnelingRateConstant({ .A_ = 1.2, .B_ = 2.3, .C_ = 302.3 }))
+                   .phase(gas_phase);
+
+  Process r8 = Process::create()
+                   .reactants({ c })
+                   .products({ yields(g, 1) })
+                   .rate_constant(UserDefinedRateConstant({.label_="my photolysis rate"}))
+                   .phase(gas_phase);
+
+  Process r9 = Process::create()
+                   .products({ yields(a, 1) })
+                   .rate_constant(UserDefinedRateConstant({.label_="my emission rate"}))
+                   .phase(gas_phase);
+
+  Process r10 = Process::create()
+                   .reactants({ b })
+                   .rate_constant(UserDefinedRateConstant({.label_="my loss rate"}))
+                   .phase(gas_phase);
+
+  auto chemical_system = System(micm::SystemParameters{ .gas_phase_ = gas_phase });
+  auto reactions = std::vector<micm::Process>{ r1, r2, r3, r4, r5, r6, r7, r8, r9, r10 };
+
+  RosenbrockSolver<Matrix, SparseMatrixPolicy> solver{ chemical_system,
+                                                       reactions,
+                                                       RosenbrockSolverParameters::three_stage_rosenbrock_parameters() };
+  State state = solver.GetState();
+
+  state.conditions_[0].temperature_ = 287.45;  // K
+  state.conditions_[0].pressure_ = 101319.9;   // Pa
+
+  state.SetConcentration(a, 1.0);  // mol m-3
+  state.SetConcentration(b, 0.0);  // mol m-3
+  state.SetConcentration(c, 0.0);  // mol m-3
+  state.SetConcentration(d, 0.0);  // mol m-3
+  state.SetConcentration(e, 0.0);  // mol m-3
+  state.SetConcentration(f, 0.0);  // mol m-3
+  state.SetConcentration(g, 0.0);  // mol m-3
+
+  state.SetCustomRateParameter("C.effective radius [m]", 1e-7);
+  state.SetCustomRateParameter("C.particle number concentration [# m-3]", 2.5e6);
+
+
+  // choose and timestep a print the initial state
+  double time_step = 500;  // s
+
+  print_header();
+  print_state(0, state);
+
+  double photo_rate = 1e-10;
+  double emission_rate = 1e-20;
+  double loss = emission_rate * 1e-3;
+  // these rates are constant through the simulation
+  state.SetCustomRateParameter("my emission rate", emission_rate);
+  state.SetCustomRateParameter("my loss rate", loss);
+
+  // solve for ten iterations
+  for (int i = 0; i < 10; ++i)
+  {
+    // Depending on how stiff the system is
+    // the solver integration step may not be able to solve for the full time step
+    // so we need to track how much time the solver was able to integrate for and continue
+    // solving until we finish
+    double elapsed_solve_time = 0;
+    state.SetCustomRateParameter("my photolysis rate", photo_rate);
+
+    while (elapsed_solve_time < time_step)
+    {
+      auto result = solver.Solve(time_step - elapsed_solve_time, state);
+      elapsed_solve_time = result.final_time_;
+      // std::cout << "solver state: " << StateToString(result.state_) << std::endl;
```

Comment:
should this be commented out?

Reply:
Yeah. I'll remove it. It's commented so that the table output is easy to convert to rst, honestly. In a future tutorial we can talk about controlling the solver and inspecting the output more fully